src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/Dependencies/IDependencyAnalyzer.cs

```diff
@@ -15,8 +16,9 @@ public interface IDependencyAnalyzer
         /// the breaking change risk of this action.
         /// </summary>
         /// <param name="project">The project whose dependencies should be analyzed.</param>
+        /// <param name="targetframeworks">The targetframworkmonikers applicable for the project.</param>
         /// <param name="state">The current analysis state which will be updated.</param>
         /// <param name="token">The token used to gracefully cancel this request.</param>
-        Task AnalyzeAsync(IProject project, IDependencyAnalysisState state, CancellationToken token);
+        Task AnalyzeAsync(IProject project, IReadOnlyCollection<TargetFrameworkMoniker> targetframeworks, IDependencyAnalysisState state, CancellationToken token);
```

Comment:
Could this be a property on the `IDependencyAnalysisState` instead?

Reply:
Does it have to be a collection? Could it just be IEnumerable?

Reply:
Added part of IDependencyAnalysisState